File: 2025/2025-01-23/fizzbuzz.rb
#
# FizzBuzzのアルゴリズムで文字列を返す
#
# @param [Integer] 数字
# @return [String] 数字かfizz, buzz, fizzbuzzの文字列
#
def fizzbuzz(n)
  case
  when n % 15 == 0
    'fizzbuzz'
  when n % 3 == 0
    'fizz'
  when n % 5 == 0
    'buzz'
  else
    n.to_s
  end
end

def run
  1.upto(100) do |n|
    p fizzbuzz n
  end
end

run if __FILE__ == $0

File: 2025/2025-02-25/fizzbuzz.py
def fizzbuzz(param: int) -> str:
    match (param % 3, param % 5):
        case (0, 0):
            result = "FizzBuzz"
        case (0, _):
            result = "Fizz"
        case (_, 0):
            result = "Buzz"
        case _:
            result = str(param)

    return result
